hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTable.java

```diff
@@ -713,11 +714,24 @@ private void deleteInvalidFilesByPartitions(HoodieEngineContext context, Map<Str
           final HoodieStorage storage = metaClient.getStorage();
           LOG.info("Deleting invalid data file=" + partitionFilePair);
           // Delete
-          try {
-            storage.deleteFile(new StoragePath(partitionFilePair.getValue()));
```

Comment:
the storage fs has param can control retry, `hoodie.filesystem.operation.retry.enable`, you can try it

Reply:
Thanks for reminder, however it does not suitable for resolving this case.
Because retry is needed not only  when exception is thrown out, but also `fs.delete` return false.

Reply:
got it, thanks, It seems that there might be similar issues in several places. whether need to handle such scenarios uniformly in the delete API within the `HoodieRetryWrapperFileSystem` to make code clean.

Reply:
I think it's a bug, not matter whether enable `hoodie.filesystem.operation.retry.enable` or not, we should not allow delete the  file failed quietly, otherwise wrong result might happen.

Reply:
> I think it's a bug, not matter whether enable `hoodie.filesystem.operation.retry.enable` or not, we should not allow delete the file failed quietly, otherwise wrong result might happen. So change in `HoodieRetryWrapperFileSystem ` is not enough.

yes, I know this is bug, so I mean `HoodieRetryWrapperFileSystem delete api` can also return flag boolean in this code.
```java
@Override
  public boolean delete(Path f, boolean recursive) throws IOException {
    return new RetryHelper<Boolean, IOException>(maxRetryIntervalMs, maxRetryNumbers, initialRetryIntervalMs, retryExceptionsList)
        .tryWith(() -> fileSystem.delete(f, recursive)).start().booleanValue();
  }
```
may be we can fix it in `HoodieRetryWrapperFileSystem` like this, but it's global
```java
@Override
  public boolean delete(Path f, boolean recursive) throws IOException {
    return new RetryHelper<Boolean, IOException>(maxRetryIntervalMs, maxRetryNumbers, initialRetryIntervalMs, retryExceptionsList)
        .tryWith(() -> {
          boolean success = fileSystem.delete(f, recursive);
          if (!success) {
            if (fileSystem.exists(f)) {
              throw new HoodieIOException("Failed to delete invalid path: " + f);
            }
          }
          return success;
        }).start().booleanValue();
  }
```

Reply:
Agree.
But we should not ignore the cases which disable `hoodie.filesystem.operation.retry.enable`.